R/simulation.R

```diff
@@ -4,6 +4,7 @@ MODEL_CORE_SIMULATION_EXTENSIONS <- "OSPSuite.Core.Domain.ModelCoreSimulationExt
 #' @docType class
 #' @description  An OSPSuite simulation
 #' @format NULL
+#' @keywords internal

```

Comment:
What does internal means? Why aren't we exposing any of those classes for users to use if they wish?

Reply:
That's a good question, and whoever wrote these functions can answer it. I don't know why they are not exported currently. 

Note that since these functions haven't had `#' @export` tags so far, these functions have thus far been **not** available to the users.

Of course, if you want, I can add export tags to all `R6` classes we currently have, then all of them will become available in `NAMESPACE`.

Reply:
I personally think we should be exporting all of these classes, but wasn't sure if it was deliberately left internal, so I didn't export them.

Reply:
Aren't they exported by default? We can create those instances from REportingEngine.... that's why we assumed that this was the default behavior. 
I agree with you. All classes should be exported, unless they are not intended for public use but I do not see any at this moment
So what is the default behavior of an R6 class in a package? Internal, exporter or sthg else?

Reply:
**Everything** that doesn't have `#' @export` roxygen2 tag is internal. 

So anything (R6 classes, functions, lists, etc.) that doesn't have this tag is currently only internal and not user visible.

Reply:
What do you mean with user visible.Clearly those classes can be used from other packages.. (OSPSuite.ReportingEngine or even OSPSUite.PArameterIdentification). Is this a matter of documentaton not available? Sorry if this is obvious but it is not for me. I can use those classes in a script even though they are internal

Reply:
The non-exported objects will still be accessible via the `:::` operator, but those objects will not be in the package `NAMESPACE`.

This distinction is not that important if you never intend to get the package on CRAN, but it is otherwise.

Objects which are in the `NAMESPACE` are the only objects users are supposed to use, and the developer provides no guarantee that the non-exported functions will continue to work in the future releases. For example, if I decide to remove an object which was never exported, it is **not** considered a breaking change because it was never supposed to be used and was for internal purposes only.

For example, the `Printable` object is not exported by `{ospsuite}`, so when `{esqlabsR}` package needs to use this object, it [must use](https://github.com/esqLABS/esqlabsR/blob/a17c845025145b3bf51f3d59f24f8df466718fca/R/Plotable.R#L8) the `:::`, and not `::` operator because `Printable` is not in `{ospsuite}` package's `NAMESPACE`.

Reply:
I see. Still. you can use the object from a returned value (sim <- loadSimulation(...)) and use sim in your code (no :: or :::) here. I agree with you. Almost all R6 class should be exporter, specifically if they are part of exported API methods.